src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicConnection.cs

```diff
@@ -61,30 +63,79 @@ internal sealed class State
                 SingleReader = true,
                 SingleWriter = true,
             });
+
+            public void RemoveStream(MsQuicStream stream)
+            {
+                bool releaseHandles;
+                lock (this)
+                {
+                    StreamCount--;
```

Comment:
Can we add an assert here to catch negative values

Reply:
good idea. updated.